src/openrct2/ride/RideRatings.cpp

```diff
@@ -220,6 +220,12 @@ static void ride_ratings_update_state_2()
             continue;
         if (tileElement->base_height != z)
             continue;
+        if (tileElement->AsTrack()->GetRideIndex() != ride->id)
+        {
+            // Only check that the track belongs to the same ride if ride does not have buildable track
+            if (!ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_HAS_TRACK))
```

Comment:
I can't quite wrap my head around why this solves the problem in #10376 - after all, this will only affect the Fried Rice Noodles Stall, which begs the question: why on earth does it even try to generate ratings for it?

Reply:
I guess because upkeep calculation is part of ride ratings and shops have upkeep.

Reply:
You actually want to find a save where someone has mixed ride types for a flat ride. That is all track for a "flat ride" has mixed ride IDs. This is probably quite rare though.